packages/expo-dev-launcher/ios/EXDevLauncherRecentlyOpenedAppsRegistry.swift

```diff
@@ -18,33 +19,70 @@ public class EXDevLauncherRecentlyOpenedAppsRegistry: NSObject {
   }
 
   @objc
-  public func appWasOpened(_ url: String, name: String?) {
-    var registry = appRegistry
-    var appEntry: [String: Any] = ["timestamp": getCurrentTimestamp()]
-    if name != nil {
-      appEntry["name"] = name
+  public func appWasOpened(_ url: URL, manifest: EXManifestsManifestBehavior?) {
+    var appEntry: [String: Any] = [:]
+    
+    let urlAsString = url.absoluteString
+    let timestamp = getCurrentTimestamp()
+        
+    var isEASUpdate = false
+    
+    if let host = url.host {
+      isEASUpdate = host == "u.expo.dev" || host == "staging-u.expo.dev"
+    }
+    
+    appEntry["isEASUpdate"] = isEASUpdate
+    
+    if (isEASUpdate) {      
+      if let updateMessage = getQueryStringParameter(url: url, param: "updateMessage") {
+        appEntry["updateMessage"] = updateMessage
+      }
+    }
+    
+    if let manifest = manifest {
+      appEntry["name"] = manifest.name()
+      
+      let json = manifest.rawManifestJSON()
```

Comment:
not for the metadata portion unless we update the manifest package

Reply:
Gotcha. I think that would be preferred (since the main point of the manifests package is to insulate everything else from the exact object structure) but could easily be a separate follow-up task.